core/src/main/java/org/infinispan/configuration/cache/TakeOfflineConfigurationBuilder.java

```diff
@@ -26,8 +27,22 @@ public TakeOfflineConfigurationBuilder(ConfigurationBuilder builder, BackupConfi
    }
 
    /**
-    * The minimal number of millis to wait before taking this site offline, even in the case 'afterFailures' is reached.
-    * If smaller or equal to 0, then only 'afterFailures' is considered.
+    * The minimal number of milliseconds to wait before taking this site offline.
+    * <p>
+    * The switch to offline status happens after a failed request (times-out or network failure) and {@code
+    * minTimeToWait} is already elapsed.
+    * <p>
+    * When a request fails (after a successful request) the timer is set and it is stopped and reset after a successful
+    * request.
+    * <p>
+    * It defaults to 0. A zero or negative value will disable any waiting time and use only {@link
+    * #afterFailures(int)}.
+    * <p>
+    * If {@link #afterFailures(int)} and {@link #minTimeToWait(long)} are both negative or zero, the site is never taken
+    * offline.
+    * <p>
+    * If {@link #afterFailures(int)} and {@link #minTimeToWait(long)} are both positive, both condition must happen
+    * before taken the site offline.
```

Comment:
I meant to add the link to the documentation as a replacement for the pseudo-table here, having different ways of saying *almost* the same thing is just confusing IMO.

```
    * The minimal number of milliseconds to wait before taking this site offline. It defaults to 0 (disabled).
    * <p>
    * It defaults to 0. A zero or negative value will disable any waiting time and use only {@link
    * #afterFailures(int)}.
```